drivers/interrupt_controller/intc_nxp_irqsteer.c

```diff
@@ -265,6 +265,18 @@ LOG_MODULE_REGISTER(nxp_irqstr);
 #define DISPATCHER_REGMAP(disp) \
 	(((const struct irqsteer_config *)disp->dev->config)->regmap_phys)
 
+#if defined(CONFIG_XTENSA)
+#define irqsteer_level1_irq_enable(irq)     xtensa_irq_enable(XTENSA_IRQ_NUMBER(irq))
+#define irqsteer_level1_irq_disable(irq)    xtensa_irq_disable(XTENSA_IRQ_NUMBER(irq))
+#define irqsteer_level1_irq_is_enabled(irq) xtensa_irq_is_enabled(XTENSA_IRQ_NUMBER(irq))
+#elif defined(CONFIG_ARM)
+#define irqsteer_level1_irq_enable(irq)     arm_irq_enable(irq)
```

Comment:
why can't this just be? :
```suggestion
#define irqsteer_level1_irq_enable(irq)     irq_enable(irq)
```

Reply:
There may be some misunderstanding for you. Let me clarify.

irq_enable() should be irq control for whole soc covering multi-level interrupts which is used everywhere in drivers.
The calling is irq_enable() -> arch_irq_enable().

What irqsteer driver did is to define the custom irq_enable/arch_irq_enable with z_soc_irq_enable() to support multi-level interrupts.
So, irq_enable() can not be used. The driver needs level 1 interrupt control API to implement z_soc_irq_enable().

See current logic in arm irq.h.

```
#if !defined(CONFIG_ARM_CUSTOM_INTERRUPT_CONTROLLER)
extern void arch_irq_enable(unsigned int irq);
#else
void z_soc_irq_init(void);
#define arch_irq_enable(irq)            z_soc_irq_enable(irq)
#endif
```

What we need and the patch did is to change to the below logic.
Otherwise, irq_enable is forced to NVIC/GIC APIs which is not able to support multi-level interrupt.

```
#if !defined(CONFIG_ARM_CUSTOM_INTERRUPT_CONTROLLER)
extern void arch_irq_enable(unsigned int irq);
#endif

#if defined(CONFIG_ARM_CUSTOM_INTERRUPT_CONTROLLER) || defined(CONFIG_MULTI_LEVEL_INTERRUPTS)
void z_soc_irq_init(void);
#define arch_irq_enable(irq)            z_soc_irq_enable(irq)
#endif
```

Thanks.

Reply:
@wearyzen ,
I understand you. But that's not workable.
Level 2 interrupt control is finally through controlling the corresponding level 1 interrupt. 
So, here z_soc_irq_level2_enable(irq) is nothing different with z_soc_irq_enable(irq) which still needs arm APIs.

```
void arch_irq_enable(unsigned int irq)
{
#ifdef CONFIG_MULTI_LEVEL_INTERRUPTS
        if (irq_get_level(irq) > 1)
                z_soc_irq_level2_enable(irq);
        else
#endif
        NVIC_EnableIRQ((IRQn_Type)irq);
}
```

So, I think current PR is okay to privide flexibility to soc.
Other intc drivers can still use irq_nextlevel.h, but that's not the case of irqsteer driver.
Thanks.

Reply:
The multi-level interrupt support is already hard to deal with as it is. Introducing more architecture-specific APIs will only make things worse.

I believe that what this PR is doing is attempt to replicate what has been done for the XTENSA architecture for multi-level interrupts: provide functions for dealing with level 1 interrupts so that the aggregator's driver won't have to redefine them.

I understand that changing the architecture code may not be a good idea, but, at minimum, we still need to introduce a new condition in the architecture layer such that if CONFIG_MULTI_LEVEL_INTERRUPTS is enabled, we use the `z_soc_irq_*` suite (like it's done in the case of `CONFIG_ARM_CUSTOM_INTERRUPT_CONTROLLER`).

If the change proposed by @yangbolu1991 is not acceptable (i.e: have the level1 interrupt handling done by the z_arm_* suite) then maybe we can just do the level1 interrupt handling inside our driver?

e.g: for NVIC-based cores, define `irqsteer_level1_irq_enable` as `NVIC_EnableIRQ`. Same for GIC-based cores.

This is not ideal either since we'd need to introduce some additional checks but ATM I don't see another option?

Also, maybe I'm wrong here but I don't see much point to the `irq_nextlevel` API here? You'd still need to define the `z_soc_irq_*` functions?

Reply:
There is req created [here](https://github.com/zephyrproject-rtos/zephyr/issues/16168) to add multi level interrupt support with another driver but this hasn't been picked up yet so I was trying to see if we can have something that works as a generic solution.
I don't have any better suggestions atm so I'll be ok to changing the names just to be consistent with how this is done for Xtensa.